src/aklite_client_ext.cc

```diff
@@ -0,0 +1,143 @@
+#include "aktualizr-lite/aklite_client_ext.h"
+
+#include <sys/file.h>
+#include <unistd.h>
+#include <boost/format.hpp>
+#include <boost/process.hpp>
+#include <boost/property_tree/ini_parser.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include "http/httpclient.h"
+#include "libaktualizr/config.h"
+#include "liteclient.h"
+#include "primary/reportqueue.h"
+
+#include "aktualizr-lite/tuf/tuf.h"
+#include "composeapp/appengine.h"
+#include "composeappmanager.h"
+#include "docker/restorableappengine.h"
+#include "ostree/repo.h"
+#include "tuf/akhttpsreposource.h"
+#include "tuf/akrepo.h"
+#include "tuf/localreposource.h"
+#include "uptane/exceptions.h"
+
+static Uptane::Target toUptaneTarget(const TufTarget& t) {
+  Json::Value target_json;
+  target_json["hashes"]["sha256"] = t.Sha256Hash();
+  target_json["custom"]["targetFormat"] = "OSTREE";
+  target_json["length"] = 0;
+  Uptane::Target target(t.Name(), target_json);
+  return target;
+}
+
+// Returns the target that should be installed, if any.
+// It might be an updated version, a rollback target, or even the currently installed target, in case we need to sync
+// apps
+GetTargetToInstallResult AkliteClientExt::GetTargetToInstall(const LocalUpdateSource* local_update_source, int version,
```

Comment:
I would consider:

1. Making `local_update_source` the last parameter.
2. Defining bitwise enumeration for  `allow_bad_target` and `force_apps_sync`, so we have a single input parameter and the target select modes can be easily extended and combined through the bitwise OR.